incremental-font-transfer/src/glyph_keyed.rs

```diff
@@ -219,6 +236,12 @@ fn synthesize_glyf_and_loca<OffsetType: LocaOffset + TryFrom<usize>>(
     new_glyf: &mut [u8],
     new_loca: &mut [u8],
 ) -> Result<(), PatchingError> {
+    if !loca.all_offsets_are_ascending() {
+        return Err(PatchingError::FontParsingFailed(ReadError::MalformedData(
+            "loca contains unordered offsets.",
```

Comment:
this is a good counter example; we want to use a string here because the particular details of how data are malformed depend on context, and it's useful to let the user add that context where needed.